mesonbuild/dependencies/base.py

```diff
@@ -308,17 +309,37 @@ def is_built(self) -> bool:
     def get_partial_dependency(self, *, compile_args: bool = False,
                                link_args: bool = False, links: bool = False,
                                includes: bool = False, sources: bool = False,
-                               extra_files: bool = False) -> InternalDependency:
-        final_compile_args = self.compile_args.copy() if compile_args else []
-        final_link_args = self.link_args.copy() if link_args else []
-        final_libraries = self.libraries.copy() if links else []
-        final_whole_libraries = self.whole_libraries.copy() if links else []
-        final_sources = self.sources.copy() if sources else []
-        final_extra_files = self.extra_files.copy() if extra_files else []
-        final_includes = self.include_directories.copy() if includes else []
+                               extra_files: bool = False, only_exts: bool = False) -> InternalDependency:
+        final_compile_args = self.compile_args.copy() if compile_args and not only_exts else []
+        final_link_args = self.link_args.copy() if link_args and not only_exts else []
+        final_libraries = self.libraries.copy() if links and not only_exts else []
+        final_whole_libraries = self.whole_libraries.copy() if links and not only_exts else []
+        final_sources = self.sources.copy() if sources and not only_exts else []
+        final_extra_files = self.extra_files.copy() if extra_files and not only_exts else []
+        final_includes = self.include_directories.copy() if includes and not only_exts else []
```

Comment:
Given the fact that I would actually expect an error if you tried to combine `only_extsd` with `whole_libraries` and `libraries`, I wonder whether it would be better to just add a new method at the API level, even if the implementation used the same interface, and then simply had an assert like `assert(not only_exts and (libraries or whole_libraries), 'only_exts conflicts with libraries or whole_libraries, this is a programming error')`. From the API level you'd just get something like `dep.pack_static_lib()` (or however you like to paint your bikeshed).